src/java/org/apache/cassandra/io/sstable/AbstractSSTableSimpleWriter.java

```
@@ -112,14 +117,15 @@ protected SSTableTxnWriter createWriter(SSTable.Owner owner) throws IOException
         SerializationHeader header = new SerializationHeader(true, metadata.get(), columns, EncodingStats.NO_STATS);
 
         if (makeRangeAware)
-            return SSTableTxnWriter.createRangeAware(metadata, 0, ActiveRepairService.UNREPAIRED_SSTABLE, ActiveRepairService.NO_PENDING_REPAIR, false, format, header);
+            return SSTableTxnWriter.createRangeAware(metadata, 0, ActiveRepairService.UNREPAIRED_SSTABLE, ActiveRepairService.NO_PENDING_REPAIR, false, MutationIdRanges.NONE, format, header);
```

In what context is this used and why can we not provide `MutationIdRanges`?

This is implemented like test code, but not stored in the test package. It should probably be moved since it looked like it was all only used for test.

If it stays here then the source of `MutationIdRanges.NONE` should be the test code constructing this so we know we don't get down here and haven't provided mutation id ranges when we should have.

---

This is used for unit tests that produce SSTables, and by the cassandra-analytics write path. I've updated to fail early with a comment indicating upcoming support via coordinated-import.